packages/webpack-plugin/src/compiler.ts

```diff
@@ -64,9 +64,16 @@ function compileVanillaSource(
       loader._compiler.webpack && loader._compiler.webpack.version,
     );
     const compat = createCompat(isWebpack5);
-    // Child compiler will compile vanilla-extract files to be evaled during compilation
-    const outputOptions = { filename: loader.resourcePath };
 
+    // Escape webpack template strings in output files so they don't get replaced
+    // Non-standard escape syntax, see the docs https://webpack.js.org/configuration/output/#template-strings
+    const outputOptions = {
+      filename: loader.resourcePath
+        .replaceAll('[', '[\\')
+        .replaceAll(']', '\\]'),
```

Comment:
What do you think about this? We probably want to test this in isolation, to make sure we're not missing any edge cases.

```suggestion
      filename: loader.resourcePath.replaceAll(
        /\[([^\[\]]*)\]/g,
        (_substring, match) => '[\\' + match + '\\]',
      ),
```

Reply:
Was about to suggest something similar. I think your one will match the inner pair of `[[]]`, but I doubt anyone would put that in a file path anyway.

Reply:
[You'd be surprised.](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#optional-catch-all-segments)

Reply:
I made it [this far](https://github.com/vanilla-extract-css/vanilla-extract/compare/webpack-square-brackets-fix...escape-webpack), but I'm not happy with how `[[blah]]` is escaped. Do you want to run some tests in Next.js?

Reply:
I'd have more confidence if we had integration tests. I've added some folders to one of the fixtures to test it. I can't quite get the escaping to work in all cases.

[foo] -> [\\foo\\] is correct
[[foo]] -> [[\\foo\\]] is correct
[[...foo]]] -> ??? No idea how to escape correctly.

Reply:
So I think we don't actually need to escape `[[...foo]]` at all, because it doesn't match [this regexp](https://github.com/webpack/webpack/blob/87660921808566ef3b8796f8df61bd79fc026108/lib/TemplatedPathPlugin.js#L19). So we only want to either escape single or double brackets, either with no contents, or some contents, but not `...`. 🥵

Reply:
This seems to work for all _four_ cases (we don't want to escape just a plain `[]` in a path). It's not quite perfect, as it won't handle `[[..foo]]` for example, but that's just incorrect syntax anyway.
```tsx
      filename: loader.resourcePath.replaceAll(
        /\[([^\[\]\.]+)\]/g,
        (_substring, match) => '[\\' + match + '\\]',
      )
```